### PR TITLE
ACS: Improve backing off to account for EOF non-errors

### PR DESCRIPTION
**Pull request by euank**
EOF isn't a real error; this changes ACS  backing off to account for that.

### CODE REVIEW

File: agent/acs/handler/acs_handler.go [hunk 1]
```diff
@@ -52,37 +53,47 @@ var SequenceNumber = utilatomic.NewIncreasingInt64(1)
...
-
-		updater.AddAgentUpdateHandlers(client, cfg, stateManager, taskEngine)
-
-		err = client.Connect()
-		if err != nil {
-			log.Error("Error connecting to ACS: " + err.Error())
-			return err
+	backoff := utils.NewSimpleBackoff(time.Second, 2*time.Minute, 0.2, 2)
+	for {
+		acsError := func() error {
```
Comment (samuelkarp) [inline agent/acs/handler/acs_handler.go]: Any particular reason you've made this an anonymous function instead of something like startSession?

Comment (euank) [inline agent/acs/handler/acs_handler.go]: I didn't see a particular reason to split it out.

Comment (samuelkarp) [inline agent/acs/handler/acs_handler.go]: I have a minor aversion to anonymous functions unless there is a compelling reason (like the desire to close over things outside the function) as I think they're a bit harder to read.  Feel free to leave it as-is if you think it's appropriate.

**Comment by samuelkarp** [comment]
:+1: 

As before, I have a minor preference against the anonymous function you have, but it's not blocking.

**Pull request merged**